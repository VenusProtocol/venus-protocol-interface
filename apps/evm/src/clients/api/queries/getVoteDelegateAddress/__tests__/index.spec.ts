import type { PublicClient } from 'viem';

import fakeAddress from '__mocks__/models/address';

import { getVoteDelegateAddress } from '..';

describe('getVoteDelegateAddress', () => {
  it('returns delegate address in the right format on success', async () => {
    const fakePublicClient = {
      readContract: async () => fakeAddress,
    } as unknown as PublicClient;

    const res = await getVoteDelegateAddress({
      publicClient: fakePublicClient,
      xvsVaultAddress: fakeAddress,
      accountAddress: fakeAddress,
    });

    expect(res.delegateAddress).toBe(fakeAddress);
    expect(res).toMatchSnapshot();
  });

  it('returns undefined when no delegate is set', async () => {
    const fakePublicClient = {
      readContract: async () => '0x0000000000000000000000000000000000000000',
    } as unknown as PublicClient;

    const res = await getVoteDelegateAddress({
      publicClient: fakePublicClient,
      xvsVaultAddress: fakeAddress,
      accountAddress: fakeAddress,
    });

    expect(res.delegateAddress).toBeUndefined();
    expect(res).toMatchSnapshot();
  });
});
