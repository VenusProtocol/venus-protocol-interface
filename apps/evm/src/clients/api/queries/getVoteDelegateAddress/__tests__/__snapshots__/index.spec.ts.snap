// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getVoteDelegateAddress > returns delegate address in the right format on success 1`] = `
{
  "delegateAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
}
`;

exports[`getVoteDelegateAddress > returns undefined when no delegate is set 1`] = `
{
  "delegateAddress": undefined,
}
`;
